dashboard/app/controllers/api_controller.rb

```diff
@@ -546,6 +547,120 @@ def assessments_structure
     render json: assessments
   end
 
+  # Return a hash by student_id for all students in a section. The value is a hash by
+  # script_level_id, with information on the student responses for that assessment.
+  # Currently, very similar logic to section_assessments, but will change as we refactor. The format
+  # of the results is different enough that I'm duplicating some code it for now.
+  # TODO(caleybrock): currently only used in internal experiment, must add controller tests.
+  def assessments_responses
```

Comment:
A couple things that would be helpful to add to your documentation of  this method:

1. The route that it implements (i.e. `GET /dashboardapi/section_assessments_responses/:section_id`)
2. Sample output - you give a paragraph-form description of the return value here but something more diagrammatic might be helpful.  Then again, that will be likely to go out-of-date soon enough, and tests would be better documentation; maybe hold off on this until tests are written?